scipy/optimize/_linprog_util.py

```diff
@@ -368,84 +374,60 @@ def _clean_inputs(lp):
-        elif (len(bounds) == 2 and np.isreal(bounds[0])
-                and np.isreal(bounds[1])):
-            bounds = [(bounds[0], bounds[1])] * n_x
-        else:
-            raise ValueError(
-                "Invalid input for linprog: exactly one lower bound and one "
-                "upper bound must be specified for each element of x")
-
-        clean_bounds = []  # also creates a copy so user's object isn't changed
-        for i, b in enumerate(bounds):
-            if b[0] is not None and b[1] is not None and b[0] > b[1]:
-                raise ValueError(
-                    "Invalid input for linprog, bound " +
-                    str(i) +
-                    " " +
-                    str(b) +
-                    ": a lower bound must be less than or equal to the "
-                    "corresponding upper bound")
-            if b[0] == np.inf:
-                raise ValueError(
-                    "Invalid input for linprog, bound " +
-                    str(i) +
-                    " " +
-                    str(b) +
-                    ": infinity is not a valid lower bound")
-            if b[1] == -np.inf:
-                raise ValueError(
-                    "Invalid input for linprog, bound " +
-                    str(i) +
-                    " " +
-                    str(b) +
-                    ": negative infinity is not a valid upper bound")
-            lb = float(b[0]) if b[0] is not None and b[0] != -np.inf else None
-            ub = float(b[1]) if b[1] is not None and b[1] != np.inf else None
-            clean_bounds.append((lb, ub))
-        bounds = clean_bounds
+        b = np.array(bounds, dtype=float)
     except ValueError as e:
-        if "could not convert string to float" in e.args[0]:
-            raise TypeError
-        else:
-            raise e
+        raise ValueError("Invalid input for linprog: unable to interpret bounds, check values and dimensions: "+e.args[0])
     except TypeError as e:
-        print(e)
-        raise TypeError(
-            "Invalid input for linprog: bounds must be a sequence of "
-            "(min,max) pairs, each defining bounds on an element of x ")
+        raise TypeError("Invalid input for linprog: unable to interpret bounds, check values and dimensions: "+e.args[0])
+
+    # Check bounds options
+    bsh = b.shape
+    if len(bsh) == 2 and np.all(bsh == (n_x, 2)):
+        # Regular N x 2 array
+        clean_bounds = b
+    elif len(bsh) == 2 and (np.all(bsh == (2, 1)) or np.all(bsh == (1, 2))):
+        # 2 values: interpret as overall lower and upper bound
+        bf = b.flatten()
+        clean_bounds[:, 0] = bf[0]
+        clean_bounds[:, 1] = bf[1]
+    elif len(bsh) == 2 and np.all(bsh == (2, n_x)):
+        # Reject a 2 x N array
+        raise ValueError("Invalid input for linprog: provide a {:d} x 2 array for bounds, not a 2 x {:d} array.".format(n_x,n_x))
+    elif len(bsh) == 1 and bsh[0] == 2:
```

Comment:
I think you can get rid of this if you use `np.atleast_2d` in the line that does `np.array(bounds, dtype=float)`.

Reply:
It would remove a branch from the if-statement, but I will have to check what the implications are for a `None`-input and a `[]`-input.